### ISSUE TITLE
 Release v3.4 patch releases discussion

### ISSUE THREAD

**Issue by terrytangyuan** [argoproj/argo-workflows#11851]
This issue tracks commits for v3.4 patch releases.

**Comment by agilgur5** [comment]
Do we want to cherry-pick in my UI date filter fixes? i.e. #11742, #11792, #11840

If so, I think I'd want to make another PR to remove the date filter default from the upgrade notes modal (as #11840 removes the default)

**Comment by terrytangyuan** [comment]
Sure I think we'll include all fixes if there's no conflict.

**Comment by terrytangyuan** [comment]
We need to include https://github.com/argoproj/argo-workflows/pull/11971 to unblock the build.

**Comment by terrytangyuan** [comment]
@juliev0 and @sarabala1979 suggests including https://github.com/argoproj/argo-workflows/pull/11774

**Comment by agilgur5** [comment]
> Sure I think we'll include all fixes if there's no conflict.

Ok, I added #11982 on top of those since they'll be included in a 3.4.x release then (the upgrade note in the modal is only for 3.4)

**Comment by terrytangyuan** [comment]
From @juliev0:
> Without it, users in a cluster with too many sso groups can't log in to Argo Server.

So we could consider that a fix.

**Comment by terrytangyuan** [comment]
Security fix https://github.com/argoproj/argo-workflows/pull/11986

**Comment by terrytangyuan** [comment]
https://github.com/argoproj/argo-workflows/pull/11993

**Comment by terrytangyuan** [comment]
Working branch https://github.com/argoproj/argo-workflows/commits/release-3.4.12

**Comment by terrytangyuan** [comment]
Tag created: https://github.com/argoproj/argo-workflows/releases/tag/v3.4.12
Images will be published once CI succeeds: https://github.com/argoproj/argo-workflows/actions/runs/6587281270/job/17897237879

**Comment by terrytangyuan** [comment]
Released

**Comment by terrytangyuan** [comment]
Reusing this issue to track any potential bug fixes for future v3.4 patch releases.

**Comment by terrytangyuan** [comment]
From @juliev0, https://github.com/argoproj/argo-workflows/commit/75bd0b83a479997da1940e048d5161b11cecb303

**Comment by terrytangyuan** [comment]
Security fix https://github.com/argoproj/argo-workflows/pull/12111

**Comment by terrytangyuan** [comment]
https://github.com/argoproj/argo-workflows/releases/tag/v3.4.13

**Comment by terrytangyuan** [comment]
TODOs:

- [ ] Revert https://github.com/argoproj/argo-workflows/commit/46297cad798cb48627baf23d548a2c7e595ed316

**Comment by agilgur5** [comment]
#12172 specifically for the 3.4.x patch series

**Comment by terrytangyuan** [comment]
https://github.com/argoproj/argo-workflows/pull/12133

**Comment by agilgur5** [comment]
#12201 as the original PR was [accidentally cherry-picked](https://github.com/argoproj/argo-workflows/pull/12201#discussion_r1393977552) into 3.4.x when it was a fix for a 3.5.x feature.

**Comment by juliev0** [comment]
Sorry, I forgot to ask before to merge this one: https://github.com/argoproj/argo-workflows/pull/11470/files (no hurry)

**Comment by terrytangyuan** [comment]
As discussed in today's contributors meeting, for v3.4 patch release, I will only cherry-pick the additional commits mentioned in this issue comments.

**Comment by terrytangyuan** [comment]
Commits to include in v3.4.14 release:

- [ ] Revert https://github.com/argoproj/argo-workflows/commit/46297cad798cb48627baf23d548a2c7e595ed316
- [ ] https://github.com/argoproj/argo-workflows/pull/12172
- [ ] https://github.com/argoproj/argo-workflows/pull/12133
- [ ] https://github.com/argoproj/argo-workflows/pull/11470

**Comment by juliev0** [comment]
requesting this be incorporated as well: https://github.com/argoproj/argo-workflows/pull/12203 (this is based on a Regression [issue](https://github.com/argoproj/argo-workflows/issues/12165) called out in 3.5.1 but I assume it also applies to 3.4.13, which is where it was noticed by someone at Intuit)

**Comment by terrytangyuan** [comment]
> requesting this be incorporated as well: #12203 (this is based on a Regression [issue](https://github.com/argoproj/argo-workflows/issues/12165) called out in 3.5.1 but I assume it also applies to 3.4.13, which is where it was noticed by someone at Intuit)

It's already patched in https://github.com/argoproj/argo-workflows/pull/12201 to release-3.4.14 branch

**Comment by terrytangyuan** [comment]
v3.4.14 tag created https://github.com/argoproj/argo-workflows/releases/tag/v3.4.14

**Comment by calvinwiebe** [comment]
May I request ![12281](https://github.com/argoproj/argo-workflows/pull/12281) be included in the next 3.4 patch? Upgrading to 3.5 caused some issues we don't have cycles to debug at the moment and would prefer to remain on 3.4 if we could. Thanks!

**Comment by juliev0** [comment]
Request to include https://github.com/argoproj/argo-workflows/pull/12353

**Comment by terrytangyuan** [comment]
This needs to be included https://github.com/argoproj/argo-workflows/pull/12470

**Comment by sarabala1979** [comment]
I am working on releasing v3.4.15 patch that includes
https://github.com/argoproj/argo-workflows/pull/12470
https://github.com/argoproj/argo-workflows/pull/12353
https://github.com/argoproj/argo-workflows/pull/12281

@terrytangyuan @agilgur5 @juliev0 Let me know if you need to include any other PRs

**Comment by sarabala1979** [comment]
@calvinwiebe https://github.com/argoproj/argo-workflows/pull/12281  has depended on another PR which has spec change. I would recommend upgrading v3.5 which is more stable now.

**Comment by agilgur5** [comment]
#12397 would be good to have for any patch

**Comment by sarabala1979** [comment]
Released https://github.com/argoproj/argo-workflows/releases/tag/v3.4.15 contains below commits
[b3aaaa45d](https://github.com/argoproj/argo-workflows/commit/b3aaaa45daff66444e42b8d78dd149e8dc132a6b) chore(build): Replaced master to main in changelogs
[fbf933563](https://github.com/argoproj/argo-workflows/commit/fbf9335635225eaa54420f3a520ef5c0043e5c34) fix: Resolve vulnerabilities in axios (#12470)
[feb992505](https://github.com/argoproj/argo-workflows/commit/feb992505ecbd57ddd16b8d27b88e26045bf3588) fix: liveness check (healthz) type asserts to wrong type (#12353)
[edb0a98c3](https://github.com/argoproj/argo-workflows/commit/edb0a98c31d7a459233131ae11e783ee45301ce6) fix(docs): release-3.4 readthedocs backport (#12474)

**Comment by sarabala1979** [comment]
> #12397 would be good to have for any patch

this commit requires other UI PR need to be cherry-picked which has UI file conflicts.
Can you ask user to pick v3.5?

**Comment by terrytangyuan** [comment]
We might need another patch ASAP https://github.com/argoproj/argo-workflows/pull/12515

**Comment by terrytangyuan** [comment]
Released https://github.com/argoproj/argo-workflows/releases/tag/v3.4.16

**Comment by agilgur5** [comment]
Another vuln #12556 (which is actually from a new `go-git` dep after #12515)

**Comment by agilgur5** [comment]
> > #12397 would be good to have for any patch
> 
> this commit requires other UI PR need to be cherry-picked which has UI file conflicts. Can you ask user to pick v3.5?

Yea this was my bad, it appears to only apply to 3.5 actually (I didn't realize that as it wasn't the Workflow List but the Workflow Details page, apparently that too was merged in 3.5). I edited my comment. Sorry about that!

**Comment by agilgur5** [comment]
#12492 would be good for any patch release

**Comment by tachylatus** [comment]
Could you please include #11847 in the next patch release?
Our workflow-controller is susceptible to crashing with "was unable to obtain node for ...".

**Comment by agilgur5** [comment]
#12622 is a workaround for a k8s 1.27 issue (#11657)

**Comment by agilgur5** [comment]
#12652 adds some missing docs that are still relevant to `release-3.4`

**Comment by agilgur5** [comment]
If it applies neatly, #12666 fixes a warning and performance issue

**Comment by agilgur5** [comment]
#12609 fixes a regression dating back to 3.3, so would be good to get into 3.5.x and 3.4.x
#12581 is a docs clarification that would be helpful to backport

**Comment by jackivanov** [comment]
https://github.com/argoproj/argo-workflows/pull/12622 impact is significant, please release as soon as possible

**Comment by agilgur5** [comment]
#12683 fixes a regression from 3.4.7+

**Comment by isubasinghe** [comment]
Sorry for the delay, this took a bit more time than I thought, I was a bit sick last week so couldn't finish it. I've only just finished merging everything in.  


Please help me check for correctness: https://github.com/isubasinghe/argo-workflows/tree/release-3.4.17

I will go through the commits again and confirm conflicts have been resolved correctly, will release tomorrow night AEDT if everything is okay.

**Comment by terrytangyuan** [comment]
Why not use release-3.4 branch? See the first paragraph in https://github.com/argoproj/argo-workflows/blob/main/docs/releasing.md

**Comment by isubasinghe** [comment]
I didn't want any of these changes to be considered official until correctness had been verified.

**Comment by ishivanshgoel** [comment]
@isubasinghe could you kindly confirm if version 3.4.17 has been released?

**Comment by terrytangyuan** [comment]
@ishivanshgoel You can find all the releases here: https://github.com/argoproj/argo-workflows/tags

**Comment by ishivanshgoel** [comment]
@tachylatus sure, thanks for the reply. But I couldn't find any recent 3.4 updates after 3.4.16 in January. 
Is there any plan to release an update with the vulnerability patches?

**Comment by Joibel** [comment]
Can we get #11585 in to this as well? Would you like me to make a PR for a fixed up version of it?

**Comment by terrytangyuan** [comment]
> Can we get #11585 in to this as well? Would you like me to make a PR for a fixed up version of it?

Yes please.

**Comment by tooptoop4** [comment]
can https://github.com/argoproj/argo-workflows/pull/12233 be included?

**Comment by terrytangyuan** [comment]
https://github.com/argoproj/argo-workflows/releases/tag/v3.4.17

**Comment by terrytangyuan** [comment]
UI security fix: https://github.com/argoproj/argo-workflows/pull/13069

**Comment by agilgur5** [comment]
> https://github.com/argoproj/argo-workflows/releases/tag/v3.4.17

@terrytangyuan @isubasinghe It looks like some fixes to 3.5 features landed in 3.4.17, which aren't applicable to 3.4. Please see my detailed comments in https://github.com/argoproj/argo-workflows/pull/13043#pullrequestreview-2072217531

**Comment by agilgur5** [comment]
> UI security fix: #13069

This would require a very significant amount of effort to backport as I detailed in https://github.com/argoproj/argo-workflows/pull/13069#pullrequestreview-2079815858.
Argo v3.4 is [on `swagger-ui-react` v4](https://github.com/argoproj/argo-workflows/blob/89cbdb53361cbe59fbe81b887ee82722cce5de54/ui/package.json#L41) as #12540 was never backported and then there are several React incompatibilities with that PR itself as well.

**Comment by terrytangyuan** [comment]
https://github.com/argoproj/argo-workflows/pull/12383 is requested to be cherry-picked by community contributor

**Comment by isubasinghe** [comment]
#13553 should also go in 3.4 I think, it fixes an old issue to do with deadlocks

**Comment by agilgur5** [comment]
#13820 fixes a regression in 3.4.8+

**Comment by klayhamn** [comment]
Hello, what are the chances of applying this fix:

https://github.com/argoproj/argo-workflows/commit/6abe8a9503dcde3de7057922c6d7688c68ae4957#diff-24284b831e81f770335038e710187e2feb0c7c3f46148c31fea108516ce1ae59R25

which seems to fix:
https://github.com/argoproj/argo-workflows/issues/8958

in 3.4?
my company currently uses 3.4.x and the RC the fix is on (3.6.x) is still a way off

**Comment by terrytangyuan** [comment]
@klayhamn That is a feature so I don't think we'll include in 3.4 patch.

**Comment by isubasinghe** [comment]
Given 3.4 is out of maintenance, I will be closing this issue.
